Clarify how default scope settings can affect all site-accesses, including admin (#1212)
ezsystems/developer-documentation: docs/guide/templates.md

This is what individual keys in the configuration mean:

- `ezpublish` and `system` are obligatory at the start of any configuration file which defines views.

> - `site_group` defines the scope for which the configuration will be used. `site_group` is a SiteAccess group used out of the box for all front pages.

Reviewer: ```suggestion
- `site_group` defines the scope for which the configuration will be used. `site_group` is a SiteAccess group used out of the box for all frontend siteacceses.
```

Just more specific terminology here

- `ezpublish` and `system` are obligatory at the start of any configuration file which defines views.
- `default` defines the SiteAccess for which the configuration will be used. "default", as the name suggests, determines what views are used when no other configuration is chosen. You can also have separate keys defining views for other SiteAccesses.

> Be aware that the `default` value will override back and front views. It may be replaced by the desired siteaccess. See example:
> 
> ``` yaml
> # Sample configuration file overriding your default siteaccess
> ezpublish:
>     system:
>         your_default_siteaccess:

Reviewer: If we want to show what config is wrong, then I'd write it like:

```suggestion
Be aware that the `default` value will be used by both frontend and backend. It may be replaced by the desired siteaccess. See example:

``` yaml
# Sample configuration file overriding a file in default scope, used by all siteaccesses including admin:
ezpublish:
    system:
        default:
```

If the point is to show a better approach then maybe rather:
```suggestion
Be aware that the `default` value will override back and front views. If you are looking to apply to only frontend, use a shared siteaccess group for that. See example:

``` yaml
# Sample configuration file for setting content view matching for a shared siteaccess group:
ezpublish:
    system:
        frontend_group:
```

Besides this, @DominikaK maybe this needs to go into a caution block or similar to be better visible in the middle of this text?

Reviewer 2: I'm actually not sure what this example is supposed to illustrate. If the intention is to say: "use a specific SA instead of default", we don't need a full example, because the "Be aware" note will be enough.
What we can (and probably should) do is replace the default key in the main example above (line 39) with another scope or a `<siteaccess>` placeholder

Author: @DominikaK Yes, the example is for that illustration you mentioned, but the reason why this is relevant on top of the note is that most of developers are used to copy paste codes examples blocs without taking the time to read details' subtleties.
Also, keeping both examples shows a global scope of the overriding system.
I suggest to keep the second suggestion from @andrerom

Reviewer 2: OK, so to avoid copy-pasting errors, it's even more important to make it an example that will not break the Back Office.

How about changing the example to refer to `site_group` (then we'll need to change the description to "`site_group` defines the SiteAccess for which the configuration will be used...", etc.)

While an example of using `default` should land in https://doc.ibexa.co/en/master/guide/siteaccess/#scope, where we already have an example for `global`.

Reviewer: I'm with @DominikaK here, the whole section above should be updated, even the bullet point from before is actually wrong here: ``default` defines the SiteAccess for which the configuration will be used. (...)`, as it kind of implies default is a site-access when it's a config scope

Would you be able to give it a go @DominikaK? As you seem to have a clear idea how you'd like to present this ;)

Author: Yes @DominikaK , this way sounds good to me. Would you mind to suggest it as a change, then I will commit it, or if you have a better way to proceed on this.

Reviewer 2: @lionelakpagni I couldn't comment with suggestion to a different page, so please see my last commit.
